### PR TITLE
Feature/#2979 add violation modal, drag-n-drop directive

### PR DESCRIPTION
**Pull request by oliachipko**
Screenshot_8

### CODE REVIEW

**Review by tkhristina — APPROVED** [review]

**Review by rostikdoroshenko — APPROVED** [review]

**Review by YuriShevchuk — APPROVED** [review]

**Review by Edw1x — APPROVED** [review]

File: src/app/ubs-admin/components/add-violations/add-violations.component.html [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  <div class="row justify-content-end mb-5">
+    <button *ngIf="images[0]?.src" (click)="deleteAllImages()" class="delete-photo">
+      <img src="assets/img/icon/bin.svg" alt="delete-icon" class="delete-icon" /> {{ 'add-violation-modal.delete-button' | translate }}
+    </button>
+    <button class="cancel-button" [mat-dialog-close]="true">{{ 'add-violation-modal.cancel-button' | translate }}</button>
+    <button class="add-button">{{ 'add-violation-modal.add-button' | translate }}</button>
+  </div>
+  <div class="row">
+    <p class="info">{{ 'add-violation-modal.date' | translate }} <span class="date">25.06.2021</span></p>
+    <p class="info">{{ 'add-violation-modal.info-about-date' | translate }} <span class="user">Current User</span></p>
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/add-violations/add-violations.component.html:59]: we should not have hardcoded text in html file

File: src/app/shared/drag-and-drop/dragDrop.directive.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    this.stopPropagation(evt, this.color.white);
+  }
+
+  @HostListener('drop', ['$event']) public onDrop(evt: DragEvent) {
+    this.stopPropagation(evt, this.color.white);
+    const files: FileHandle[] = [];
+    const droppedFiles = Array.from(evt.dataTransfer.files);
+    for (const fileItem of droppedFiles) {
+      const file = fileItem;
+      const url = this.sanitizer.bypassSecurityTrustUrl(window.URL.createObjectURL(file));
```
Comment (JimHolman1199) [inline src/app/shared/drag-and-drop/dragDrop.directive.ts:32]: https://sonarcloud.io/project/security_hotspots?id=ita-social-projects-green-city-client&pullRequest=991

File: src/app/ubs-admin/components/add-violations/add-violations.component.scss [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+  font-size: 14px;
+  margin-right: 30px;
+}
+
+.user {
+  color: var(--ubs-quaternary-dark-grey);
+  font-size: 14px;
+}
+
+input[type='file'] {
```
Comment (JimHolman1199) [inline src/app/ubs-admin/components/add-violations/add-violations.component.scss:97]: you can use bootstrap d-none class instead

Comment (oliachipko) [reply src/app/ubs-admin/components/add-violations/add-violations.component.scss:97]: done

**Review by andreikazmirovich — APPROVED** [review]

**Review by JimHolman1199 — APPROVED** [review]

**Pull request merged**